src/App.jsx

```diff
@@ -14,70 +15,86 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      {getHeader(selectedGood, setSelectedGood)}
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              key={good}
+              data-cy="Good"
+              className={
+                  good === selectedGood
+                    ? 'has-background-success-light'
+                    : ''
+                  }
             >
-              +
-            </button>
-          </td>
+              <td>
+                {getButton(good, selectedGood, setSelectedGood)}
+              </td>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+              <td data-cy="GoodTitle" className="is-vcentered">
+                {good}
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    </main>
+  );
+};
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+function getHeader(selectedGood, setSelectedGood) {
+  return (!selectedGood
+    ? <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+    : (
+      <h1 className="title is-flex is-align-items-center">
+        {`${selectedGood} is selected`}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <button
+          data-cy="ClearButton"
+          type="button"
+          className="delete ml-3"
+          onClick={() => {
```

Comment:
```suggestion
          onClick={reset}
```
Consider moving click handlers from jsx.
You can keep this bunch of code in the component, but click handlers should be moved from here

Reply:
I just did this following Misha's example in the video

Reply:
I did this following Misha’s example. It is impossible to move the handler outside the component, since you need access to the setSelectedGood function. In case I want to move the function outside jsx, but storing it in the component, will cause an error, since before rendering the element on which the handler needs to be installed will not yet be initialized!

Reply:
@Daniilart01 the mean that you need to create separate function reset and move this logic inside this function, for instance:
```
const reset = () => {
 setSelectedGood(null);
}
```
and

```
<button
   data-cy="ClearButton"
   type="button"
   className="delete ml-3"
   onClick={reset}
```
it will make a code cleaner